src/main/java/com/openlattice/chronicle/services/enrollment/EnrollmentService.java

```diff
@@ -0,0 +1,398 @@
+    }
+
+    @Override
+    public boolean isKnownParticipant( UUID organizationId, UUID studyId, String participantId ) {
+        return getParticipantEntityKeyId( organizationId, studyId, participantId ) != null;
+    }
+
+    @Override
+    public Map<FullQualifiedName, Set<Object>> getParticipantEntity(
+            UUID organizationId, UUID studyId, UUID participantEntityKeyId ) {
+        try {
+            ApiClient apiClient = apiCacheManager.prodApiClientCache.get( ApiClient.class );
+            DataApi dataApi = apiClient.getDataApi();
+
+            UUID entitySetId = edmCacheManager.getParticipantEntitySetId( organizationId, studyId );
+            if ( entitySetId == null ) {
+                logger.error( "Unable to ge participant ESID: orgId = {}, studyId = {}, participantEKID = {}",
+                        organizationId,
+                        studyId,
+                        participantEntityKeyId );
+                return ImmutableMap.of();
+            }
+            Map<FullQualifiedName, Set<Object>> entity = dataApi.getEntity( entitySetId, participantEntityKeyId );
+            if ( entity == null ) {
+                logger.error(
+                        "Unable to get participant entity: orgId  = {}, studyId = {} participantEKID = {}, participantESID = {}",
+                        organizationId,
+                        studyId,
+                        participantEntityKeyId,
+                        entitySetId );
+                return ImmutableMap.of();
+            }
+            return entity;
+
+        } catch ( ExecutionException e ) {
+            logger.error( "Unable to get participant entity: orgId = {}, studyId={}, participantEKID = {} ",
+                    organizationId,
+                    studyId,
+                    participantEntityKeyId,
+                    e );
+            return ImmutableMap.of();
+        }
+    }
+
+    @Override
+    public ParticipationStatus getParticipationStatus(
+            UUID organizationId, UUID studyId, String participantId ) {
+        logger.info( "getting participation status: orgId = {}, studyId = {}, participantId = {}",
+                organizationId,
+                studyId,
+                participantId );
+
+        try {
+            ApiClient apiClient = apiCacheManager.prodApiClientCache.get( ApiClient.class );
+            SearchApi searchApi = apiClient.getSearchApi();
+
+            // entity set ids
+            UUID studiesESID = edmCacheManager.getEntitySetId( organizationId, CHRONICLE, STUDIES, STUDY_ES );
+            UUID participatedInESID = edmCacheManager.getEntitySetId( organizationId, CHRONICLE, PARTICIPATED_IN,
+                    PARTICIPATED_IN_ES );
+            UUID participantsESID = edmCacheManager.getParticipantEntitySetId( organizationId, studyId );
+
+            checkNotNullUUIDs( Sets.newHashSet( studiesESID, participatedInESID, participantsESID ) );
```

Comment:
Because of the way we have restructured chronicle, some of the entity set ids may be null if an organization hasn't installed the corresponding app. Say for example a participant in a certain organization that has not installed the data collection component attempts to enroll their device. The device entity set id in this case would be null, and the call would fail.